### ISSUE TITLE
Generated CeresTargets.cmake contains invalid path

### ISSUE THREAD

**Issue by extrowerk** [ceres-solver/ceres-solver#649]
On
```
~> uname -a
Haiku shredder 1 hrev54812 Dec 25 2020 07:24:00 x86_64 x86_64 Haiku
```
building OpenCV against Ceres 2.0.0 (current ceres WIP port: https://github.com/haikuports/haikuports/pull/5523 ) results:
```
CMake Error in /sources-2/opencv_contrib-4.5.1/modules/rgbd/CMakeLists.txt:
  Imported target "Ceres::ceres" includes non-existent path

    "/boot/system/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

The line cmake complains about looks like this:
```
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;/boot/system/develop/headers"
```

There is no include folder on Haiku, the headers stored in `/boot/system/develop/headers`.
I assume one could safely remove the `${_IMPORT_PREFIX}/include;` part alltogether, but this theory is untested.

I grepped the whole ceres source tree searching a template or something used by cmake to generate this file, but found absolutely nothing, and i dislike the idea of sedding the file to remove that part.

**Comment by sandwichmaker** [comment]
@alexsmac can you take a look please?

**Comment by extrowerk** [comment]
Any update here?

**Comment by alexsmac** [comment]
Hi @extrowerk can you clarify what CMake variables you are setting when building for your non-standard environment to control the header (and library) install locations?  Given what you want to achieve, I assume you are specifying `CMAKE_INSTALL_INCLUDEDIR` (which defaults to `include` as per docs [here](https://cmake.org/cmake/help/latest/command/install.html#installing-targets)) is that correct?

**Comment by extrowerk** [comment]
Hi @alexsmac, as you can see in the linked PR i don't specify anything other than:
```
		-DCMAKE_INSTALL_PREFIX=$prefix \
		-DBUILD_SHARED_LIBS=ON \
		-DEigen3_DIR=`finddir B_SYSTEM_DATA_DIRECTORY`/eigen3/cmake \
		$cmakeDirArgs
```

The cmakeDirArgs will be extended by [HaikuPorter ](https://github.com/haikuports/haikuporter) (the build system, this sets up the chroot, mounts the required packages, runs the supplied commands like cmake, and then packages the installed artifacts):
https://github.com/haikuports/haikuporter/blob/226a85daf6255ad9b8528a349cea5214c36d97eb/HaikuPorter/Port.py#L1212
Basically it is something similar to GNUInstallDirs. 

HaikuPorter doesn't pass anything like ```/include```. 

Ceres takes the `/include` from somewhere, but this should not happen. The no-existent path breaks cmake, you should not assume paths. There is no guarantee a folder like this exists, the include folder name is not standardized, it doesn't have to be ```include```, this is just a legacy cruft from the *nix days. On Haiku the header files should be stored in ```/boot/system/develop/headers```. 

At least this is what i recall.

**Comment by sandwichmaker** [comment]
@alexsmac what is the status of this?

**Comment by sergiud** [comment]
@extrowerk Will `GNUInstallDirs` solve your issue?

**Comment by extrowerk** [comment]
Definitely.

Sergiu Deitsch írta 2022. feb.. 14, H-n 11:26 órakor:
